src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,95 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  setSelectedTodo: (arg0: Todo | undefined) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setErrorMessage('');
+    setIsLoading(true);
+
+    if (todo) {
+      getUser(todo?.userId)
+        .then(userFromServer => {
+          setUser(userFromServer);
+          setIsLoading(false);
+        })
+        .catch(error => {
+          setErrorMessage(error.message);
+        })
+        .finally(() => setIsLoading(false));
+    } else {
+      setErrorMessage('No selected todo');
+    }
+  }, [todo]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      {isLoading && !user
+        && <Loader />}
+      {!isLoading && !errorMessage
+        && (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${todo.id}`}
+              </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => setSelectedTodo(undefined)}
```

Comment:
```suggestion
                onClick={() => setSelectedTodo(null)}
```
it's better to explicitly say it's defined but empty value - there is difference between `null` and `undefined`